lib/Controller/SignalingController.php

```diff
@@ -340,12 +343,49 @@ private function backendRoom($room) {
 			]);
 		}
 
+		$participant = null;
 		if (!empty($userId)) {
-			// Rooms get sorted by last ping time for users, so make sure to
-			// update when a user joins a room.
-			$room->ping($userId, '0', time());
+			// User trying to join room.
+			try {
+				$participant = $room->getParticipant($userId);
+			} catch (ParticipantNotFoundException $e) {
+				// Ignore, will check for public rooms later.
+			}
 		}
 
+		if (empty($participant)) {
+			// User was not invited to the room, only allow access if public room.
+			if ($room->getType() !== Room::PUBLIC_CALL) {
+				// Return generic error to avoid leaking which rooms exist.
+				return new JSONResponse([
+					'type' => 'error',
+					'error' => [
+						'code' => 'no_such_room',
+						'message' => 'The user is not invited to this room.',
+					],
+				]);
+			}
+
+			if (!empty($userId)) {
+				$room->addUsers([
+					'userId' => $userId,
+					'participantType' => Participant::USER_SELF_JOINED,
+					'sessionId' => $sessionId,
+				]);
+			} else {
+				$room->internalEnterRoomAsGuest($sessionId);
+			}
+		} else {
+			// TODO(fancycode): This will remove the user from other rooms
+			// he is currently in. Needs to get communicated to the signaling
+			// server.
+			$room->updateUserSessionId($userId, $sessionId);
```

Comment:
Why all these changes? The session id does not change when joining/leaving a room